RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitCalibrationESProducer.cc

```diff
@@ -80,79 +76,73 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
           soa(i) = values[row][i];
       }
 
+      // @short create the ESProducer product: a SoA with channel-level calibration constants
       std::optional<hgcalrechit::HGCalCalibParamHost> produce(const HGCalModuleConfigurationRcd& iRecord) {
         auto const& moduleMap = iRecord.get(indexToken_);
-        auto const& config = iRecord.get(configToken_);
+        edm::LogInfo("HGCalCalibrationESProducer") << "produce: filename=" << filename_.fullPath().c_str();
 
         // load dense indexing
         const uint32_t nchans = moduleMap.getMaxDataSize();  // channel-level size
         hgcalrechit::HGCalCalibParamHost product(nchans, cms::alpakatools::host());
 
         // load calib parameters from JSON
         std::ifstream infile(filename_.fullPath().c_str());
-        json calib_data = json::parse(infile);
-        for (const auto& it : calib_data.items()) {  // loop over module typecodes in JSON file
-          const std::string& module = it.key();      // module typecode, e.g. "ML-F3PT-TX-0003"
-          if (module == "Metadata")
-            continue;  // ignore metadata fields
-          const auto& [ifed, imod] = moduleMap.getIndexForFedAndModule(module);
-          const uint32_t offset =
-              moduleMap.getIndexForModuleData(module);  // convert module typecode to dense index for this module
-          const uint32_t nrows =
-              calib_data[module]["Channel"].size();  // number of channels to compare with JSON arrays
-          const uint32_t nrocs =
-              config.feds[ifed].econds[imod].rocs.size();  // number of channels to compare with JSON arrays
+        json calib_data = json::parse(infile, nullptr, true, /*ignore_comments*/ true);
+        for (const auto& it : moduleMap.getTypecodeMap()) {  // loop over all module typecodes
+          std::string module = it.first;                     // module typecode, e.g. "ML-F3PT-TX-0003"
```

Comment:
can this be a `const&` ?
```suggestion
          std::string const& module = it.first;                     // module typecode, e.g. "ML-F3PT-TX-0003"
```

Reply:
Thanks, done in https://github.com/cms-sw/cmssw/pull/48137/commits/5ba71768546ae894cbffa898b9c8b1bd9cad9a5b.